datafusion/core/src/execution/context.rs

```diff
@@ -816,7 +809,7 @@ impl SessionContext {
             name,
             table_path,
             listing_options,
-            options.schema,
+            options.schema.map(|s| Arc::new(s.to_owned())),
```

Comment:
For what it is worth, I think this is making a (deep) copy of the schema. On master, `SchemaRef` is a ref counted pointer (Arc), so cloning it is less expensive

Reply:
Related to an API change (CSV and JSON - AVRO unification issue).

Reply:
https://github.com/apache/arrow-datafusion/pull/4694#discussion_r1058144779